PatientWebApplication/HealthClinic.DAL/Service/OperationService.cs

```diff
@@ -197,5 +197,46 @@ private Boolean CheckIfOperation(String stringToCheck)
             return stringToCheck.Equals("Operation");
         }
 
+        private List<Operation> GetAllOperationsByDateAndDoctor(DateTime date, int doctorId)
+        {
+            List<Operation> operations = new List<Operation>();
+            foreach (Operation operation in _operationRepository.GetOperationsForDoctor(doctorId))
+            {
+                if (date == UtilityMethods.ParseDateInCorrectFormat(operation.Date))
+                {
+                    operations.Add(operation);
+                }
+            }
+            return operations;
+        }
+
+        private List<Operation> GetAllOperationsByDateAndPatient(DateTime date, int patientId)
+        {
+            List<Operation> operations = new List<Operation>();
+            foreach (Operation operation in GetOperationsForPatient(patientId))
+            {
+                if (date == UtilityMethods.ParseDateInCorrectFormat(operation.Date))
+                {
+                    operations.Add(operation);
+                }
+            }
+            return operations;
```

Comment:
Refactor code. Replace method GetAllOperationsByDateAndPatient(DateTime date, int patientId) with Where syntax ,for example :
return GetOperationsForPatient(patientId).Where(operation => date == UtilityMethods.ParseDateInCorrectFormat(operation.Date) )